nixos/modules/config/ldap.nix

```diff
@@ -214,6 +214,9 @@ in
       if cfg.daemon.enable then nss_pam_ldapd else nss_ldap
     );
 
+    # Let pwdutils tools query nslcd through lbnss_ldap.so.
+    environment.sessionVariables.LD_LIBRARY_PATH = config.system.nssModules.path;
```

Comment:
But here it's not `nslcd` which is missing `libnss_ldap.so`, it's `glibc`'s utilities like `getent` or `passwd`, which, as far as I can understand, need to load `libnss_ldap.so` to reach `nslcd`.
Like in this 2012 bug on Ubuntu: https://bugs.launchpad.net/ubuntu/+source/libnss-ldap/+bug/1022903
Maybe those utilities can be listed and wrapped to set the correct `LD_LIBRARY_PATH`, maybe there is another mechanism for that kind of situation. I don't know. Meanwhile, a system wide `LD_LIBRARY_PATH` does the job :s

Reply:
@ju1m NixOS uses nscd and modifies its LD_LIBRARY_PATH, so we don't have to do it globally - see commit message of https://github.com/NixOS/nixpkgs/commit/e71241793671c11498c388bde8a3de2d89fde25e:

> Preferably, we would not be using nscd at all. But we need to
use it because glibc reads  nss modules from /etc/nsswitch.conf
by looking relative to the global LD_LIBRARY_PATH.  Because LD_LIBRARY_PATH
is not set globally (as that would lead to impurities and ABI issues),
glibc will fail to find any nss modules.
Instead, as a hack, we start up nscd with LD_LIBRARY_PATH set
for only that service. Glibc will forward all nss syscalls to
nscd, which will then respect the LD_LIBRARY_PATH and only
read from locations specified in the NixOS config.
we can load nss modules in a pure fashion.

You might have been affected by negative caching, fixed in https://github.com/NixOS/nixpkgs/pull/50316 .

Can you rebase on master and check if that's still an issue? It might probably also make sense to add/extend a nixos test with your testcase.

Reply:
@flokli Thanks again, I don't know for the cachin, but the `LD_LIBRARY_PATH` set only for `nscd` seems to work well.